amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/resolution/stages/ExtensionsResolutionStage.scala

```diff
@@ -317,9 +317,9 @@ class ExtensionResolutionStage(override val profile: ProfileName, override val k
                                           other: AmfArray,
                                           extensionId: String,
                                           extensionLocation: Option[String]): Unit = {
-    other.values.foreach { value =>
-      target.add(field, value)
-    }
+    val targetValues = target.fields.get(field).asInstanceOf[AmfArray].values
```

Comment:
Are we sure that the value of the `target` element in the specific `field` is always an AmfArray?

Reply:
yes, that method is called `setDomainElementArrayValue` and always sets an array value. I've just changed the array it sets to not have duplicated values.